src/ocamlutil/pretty.ml

```diff
@@ -456,16 +464,16 @@ let rec scan (abscol: int) (d: doc) : int =
 
 (** Keep a running counter of the newlines we are taking. You can read and
     reset this from user code, if you want *)
-let countNewLines = ref 0
+let countNewLines = refDLS 0
 
 (* The actual function that takes a document and prints it *)
 let emitDoc
     (emitString: string -> int -> unit) (* emit a number of copies of a
                                            string *)
     (d: doc) =
-  let aligns: int list ref = ref [0] in (* A stack of alignment columns *)
+  let aligns: int list refDLS = refDLS [0] in (* A stack of alignment columns *)
 
-  let wantIndent = ref false in
+  let wantIndent = refDLS false in
```

Comment:
Some things like this are local to a function. Do they actually escape and get transferred to other domains that they need DLS? If not, then a normal `ref` should be fine.

Reply:
[8837038](https://github.com/goblint/cil/pull/180/commits/883703823ebdc26d43137ed7d746035e24f44555)